pkg/virt-operator/deletion/all.go

```diff
@@ -0,0 +1,278 @@
+/*
+ * This file is part of the KubeVirt project
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * Copyright 2018 Red Hat, Inc.
+ *
+ */
+
+package deletion
+
+import (
+	"fmt"
+
+	extclient "k8s.io/apiextensions-apiserver/pkg/client/clientset/clientset"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/types"
+	apiclient "k8s.io/kube-aggregator/pkg/client/clientset_generated/clientset"
+
+	"kubevirt.io/kubevirt/pkg/api/v1"
+	"kubevirt.io/kubevirt/pkg/kubecli"
+	"kubevirt.io/kubevirt/pkg/log"
+)
+
+func Delete(kv *v1.KubeVirt, clientset kubecli.KubevirtClient) (int, error) {
```

Comment:
I think the delete order should be as follows:


 1) delete CRDs. As a consequence all CRs of these CRDs get marked for deletion. PATCH/PUT operations are still possible on these CRs, so our components can proceed as usual.
 2) Once the CRDs are gone, delete the rest of our infra.

Reply:
That means that you don't have to delete VMIs, VMIPresets, ... They are all gone as soon as the CRDs are gone.